src/pool.c

```diff
@@ -117,6 +118,11 @@ pool_free(Pool *pool)
   queue_free(&pool->vendormap);
   queue_free(&pool->pooljobs);
   queue_free(&pool->lazywhatprovidesq);
+  if (pool->considered)
+    {
+      map_free(pool->considered);
+      solv_free(pool->considered);
+    }
```

Comment:
We can't do that. We did not allocate pool->considered, so we do not know how to free it.

Reply:
If `pool.h` defines it as `Map *considered`, are you saying some implementations are using it as a free scratch space and not really a `Map *`?

Either way, I think the initialization fix in `pool_create()` should probably still be included.

Reply:
No, but the Map it points to can be part of some bigger struct, so we can't do a solv_free() on the pointer. E.g.:
```
struct my_pool {
  Pool *pool;
  Map considered;
};
```

There actually *is* a calloc in pool_create():
```
  pool = (Pool *)solv_calloc(1, sizeof(*pool));
```